src/ol/style/Icon.js

```diff
@@ -397,6 +397,23 @@ class Icon extends ImageStyle {
     this.normalizedAnchor_ = null;
   }
 
+  /**
+   * Get the anchor origin
+   * @return {IconOrigin} Origin
+   * @api
+   */
+  getAnchorOrigin() {
+    return this.anchorOrigin_;
+  }
+
+  /**
+   * Get the units used to express the anchor
+   * @return {[IconAnchorUnits, IconAnchorUnits]} units as [x, y]
+   */
+  getAnchorUnits() {
+    return [this.anchorXUnits_, this.anchorYUnits_];
```

Comment:
could you reuse an array here instead of allocating a new one on each call? thanks

Reply:
meaning declaring a new attribute to the Icon class? If it doesn't clutter too much the class yes.
Should I also switch the code to be using the array instead of the individual values (`this.anchorXUnits_` and `this.anchorYUnits`) so that I may remove them from the constructor, and only store these values once?

Reply:
wait, I think I need to review this a bit more; these additions might make the API confusing because `getAnchor()` always returns pixels values, regardless of the units and origin set on creation.

If you need to apply custom logic to some of the icons of your app, why not use custom properties with `set()`/`get()` instead?

Reply:
That would definitely be a useful addition, a PR would be welcome, thanks!